sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -188,26 +216,96 @@ export function bearerTokenAuthenticationPolicy(
       try {
         response = await next(request);
       } catch (err: any) {
-        error = err;
-        response = err.response;
+        if (isRestError(error)) {
+          response = err.response;
+          error = err;
+        } else {
+          throw err;
+        }
       }
 
-      if (
-        callbacks.authorizeRequestOnChallenge &&
-        response?.status === 401 &&
-        getChallenge(response)
-      ) {
-        // processes challenge
-        const shouldSendRequest = await callbacks.authorizeRequestOnChallenge({
-          scopes: Array.isArray(scopes) ? scopes : [scopes],
-          request,
-          response,
-          getAccessToken,
-          logger,
-        });
-
-        if (shouldSendRequest) {
-          return next(request);
+      async function handleCaeChallenge(claims: string): Promise<PipelineResponse> {
+        const shouldSendRequest = await authorizeRequestOnCaeChallenge(
+          {
+            scopes: Array.isArray(scopes) ? scopes : [scopes],
+            response,
+            request,
+            getAccessToken,
+            logger,
+          },
+          claims,
+        );
+        // Send updated request and handle response for RestError
+        try {
+          if (shouldSendRequest) {
+            error = undefined;
+            response = await next(request);
+          }
+        } catch (err: any) {
+          if (isRestError(error)) {
+            response = err.response;
+            error = err;
+          } else {
+            throw err;
+          }
+        }
+        return response;
+      }
```

Comment:
We do this `try { ... } catch(e) { if(isRestError(e)) { ... } }` thing a few times. Maybe a helper could be handy, something like

```ts
/**
 * Try to send the given request. 
 * 
 * When a response is received, returns a tuple of the response received and, if the response was received
 * inside a thrown RestError, the RestError that was thrown.
 * 
 * Otherwise, if an error was thrown while sending the request that did not provide an underlying response, it
 * will be rethrown.
 */
async function trySendRequest(request: PipelineRequest): [PipelineResponse, RestError | undefined] {
  try {
    return [await next(request), undefined];
  } catch(e: any) {
    if(isRestError(e) && e.response) {
      return [e.response, e];
    } else {
      throw e;
    }
  }
}
```

Then this logic would simplify to

```suggestion
if(shouldSendRequest) {
  [response, error] = await trySendRequest(request);
}
```